test/extensions/filters/http/proto_api_scrubber/scrubbing_util_lib/field_checker_test.cc

```diff
@@ -141,6 +141,57 @@ class FieldCheckerTest : public ::testing::Test {
                 }
               }
             }
+            request_field_restrictions: {
```

Comment:
Changes LGTM, but please refactor `getDefaultProtoConfig()` in a future PR.
Currently it is a big blob of change, and it is not clear which parts are responsible for each test, and hinders maintainability.